internal/instructionmng/instructionmanager.go

```diff
@@ -0,0 +1,90 @@
+package instructionmng
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jinzhu/gorm"
+
+	"github.com/filanov/bm-inventory/internal/common"
+	"github.com/filanov/bm-inventory/models"
+	logutil "github.com/filanov/bm-inventory/pkg/log"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+type InstructionManager struct {
+	log        logrus.FieldLogger
+	db         *gorm.DB
+	stateSteps map[string][]CmdStepAPI
+}
+
+type InstructionMngApi interface {
+	GetNextSteps(ctx context.Context, host *models.Host) (models.Steps, error)
+}
+
+func NewInstructionManager(log logrus.FieldLogger, db *gorm.DB) *InstructionManager {
+	connectivityCmd := NewConnectivityCheckCmd(log, db)
+	hwCmd := NewHwInfoCmd(log)
+	executeCmd := NewExecuteCmd(log)
+
+	// return &InstructionManager{
+
+	i := InstructionManager{
+		stateSteps: map[string][]CmdStepAPI{
+			common.HostStatusKnown:        {connectivityCmd},
+			common.HostStatusInsufficient: {connectivityCmd},
+			common.HostStatusDisconnected: {connectivityCmd},
+			common.HostStatusDiscovering:  {hwCmd, connectivityCmd},
+			common.HostStatusInstalling:   {executeCmd},
+		},
+	}
+	i.log = log
+	i.db = db
+	return &i
+
+}
+
+func (i *InstructionManager) GetNextSteps(ctx context.Context, host *models.Host) (models.Steps, error) {
+
+	log := logutil.FromContext(ctx, i.log)
+	ClusterID := host.ClusterID
+	HostID := host.ID
+	HostStatus := swag.StringValue(host.Status)
+
+	log.Infof("GetNextSteps cluster: ,.%s host: <%s>, host status: <%s>", ClusterID, HostID, HostStatus)
+
+	returnSteps := models.Steps{}
+
+	if cmdsMap, ok := i.stateSteps[HostStatus]; ok {
+		//need to add the step id
+		for _, cmd := range cmdsMap {
+			var copyStep = &models.Step{}
+			var err error
+			copyStep, err = cmd.GetStep(ctx, host)
```

Comment:
```suggestion
			step, err := cmd.GetStep(ctx, host)
```
then you will not need to define `var err` and `var step`

Reply:
fixed